x-pack/plugin/apm-data/src/main/resources/ingest-pipelines/traces-apm@pipeline.yaml

```diff
@@ -30,10 +30,10 @@ processors:
     field: ["event.duration"]
     ignore_failure: true
     ignore_missing: true
-- set:
+- script:
     if: ctx.event?.outcome == 'success'
-    field: event.success_count
-    value: 1
+    source: |
+        ctx.event.success_count = ctx[ctx.processor?.event]?.representative_count;
```

Comment:
I can't think of a reason why representative count would be missing, but I still think a null check feels safer.

Reply:
(Not a response to @carsonip, just a comment on the same section of the code.) What eventually happens with the value of the `event.success_count`? Does it make sense to pop a `null` in there or will that blow up later processing?

Reply:
Looking at apm-server source code, event.success_count can be nil: https://github.com/elastic/apm-server/blob/79c4599ab9bd65e1cc688eed1169505933f6f405/systemtest/ingest_test.go#L177

It is only non-nil when outcome is success or failure, in which case, non-0 and 0 respectively.

I assume it will be used by APM UI for getting a transaction success rate.